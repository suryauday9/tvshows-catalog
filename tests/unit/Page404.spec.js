// Components
import Page404 from '@/views/Page404.vue';

// Utilities
import { initializeApp } from './initializeApp';
import { createLocalVue, shallowMount } from '@vue/test-utils';

const localVue = initializeApp(createLocalVue())

describe('Page404 view', () => {
  let wrapper;
  const mountComponent = (options) => {
    return shallowMount(Page404, {
        localVue,
        ...options,
    })
  };

  beforeEach(() => {
      wrapper = mountComponent();
  })

  afterEach(() => {
      wrapper.destroy();
  })

  it('renders component correctly', () => {
      // see if component renders
      expect(wrapper.element).toMatchSnapshot();
  });

  it('should be a Vue instance', () => {
      expect(wrapper.isVueInstance).toBeTruthy();
  });

})
