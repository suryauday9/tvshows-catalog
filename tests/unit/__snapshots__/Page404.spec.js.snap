// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page404 view renders component correctly 1`] = `
<v-container-stub
  class="text-center white--text"
  tag="div"
>
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="12"
      tag="div"
    >
      <h1
        class="pt-6 black--text"
      >
        404 - Page not found
      </h1>
       
      <router-link-stub
        ariacurrentvalue="page"
        event="click"
        tag="a"
        title="Go to homepage"
        to="/"
      >
        <v-btn-stub
          activeclass=""
          class="mt-4"
          color="primary"
          tag="button"
          type="button"
        >
          Go to home
        </v-btn-stub>
      </router-link-stub>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;
